src/welcome.rs

```diff
@@ -126,6 +126,72 @@ pub(crate) fn assign_talk_role(cx: &Context, reaction: &Reaction) -> Result<(),
     Ok(())
 }
 
+pub(crate) fn update_welcome_message(args: Args) -> Result<(), Error> {
+    use std::str::FromStr;
+
+    let new_message = &args
+        .params
+        .get("message")
+        .ok_or("unable to retrieve message param")?;
+
+    let conn = DB.get()?;
+
+    let (msg, me) = conn
+        .build_transaction()
+        .read_only()
+        .run::<_, Box<dyn std::error::Error>, _>(|| {
+            let msg: Option<_> = messages::table
+                .filter(messages::name.eq("welcome"))
+                .first::<(i32, String, String, String)>(&conn)
+                .optional()?;
+
+            let me: Option<_> = users::table
+                .filter(users::name.eq("me"))
+                .first::<(i32, String, String)>(&conn)
+                .optional()?;
+
+            Ok((msg, me))
+        })?;
+
+    if let Some((_, _, cached_message_id, cached_channel_id)) = msg {
+        if let Some((_, _, user_id)) = me {
+            let mut custom_message = CustomMessage::new();
```

Comment:
Why are we using a CustomMessage here? Can we not GET then edit? Or alternatively, start caching the message/channel IDs of the welcome message?